Add get/setVehicleWheelScale

AlexTMjugador opened the pull request:
These functions, related to issue #719, simply expose the wheel scale size attribute of the `CVehicle` class to clientside Lua scripts.

## Considerations
### Why?
As explained in the related issue above, custom vehicle models may have a different wheel size than the default model they replace. This wouldn't be an issue if GTA: SA somehow calculated ride height, wheel hitbox and possibly other data from the information contained in the model, but it doesn't. Singleplayer vehicle modifications handle differing wheel sizes by modifying the `WheelScale_Front` and `WheelScale_Rear` fields in `data/vehicles.ide`, which tell GTA the size of the front and rear wheel components so that it can calculate things properly, but script developers can't change those.

On the other hand, most replacement vehicle models with differing wheel sizes that I've seen are not too far off from the sizes of the default models. I believe that the most visible effect of this discrepancy is that vehicle mods look "sunken" into the ground in MTA: SA, while in singleplayer they don't. The `setVehicleWheelScale` function that this PR adds, while **it doesn't affect how GTA: SA calculates the mentioned physical parameters for the model**, allows to draw all the wheels of a vehicle taller or shorter, so that these small differences can be compensated and the custom model will look better. It may also add the possibility of enhancing car tunning scripts.

### Why not?
Admittedly, these functions expose relatively unflexible functionality to scripts. Ideally, we should let scripts set the `WheelScale_Front` and `WheelScale_Rear` values independently, or even improve upon that and let them change the size of each wheel somehow, updating the physical parameters of the model too. However, I've heard that doing that is fairly complex, I don't really have the skills to successfully do a deep dive in GTA: SA internals, and maybe these functions are "good enough".

### Isn't this PR already covered by `setVehicleComponentScale`?
It's not! For example, consider this runcode command:
```lua
crun for _, comp in pairs({ "wheel_lf_dummy", "wheel_rf_dummy", "wheel_lb_dummy", "wheel_rb_dummy" }) do localPlayer.vehicle:setComponentScale(comp, 2, 2, 2) end
```
Intuitively, running that while in a car should make the wheels 2x bigger, but it doesn't.

### Why not fix `setVehicleComponentScale` instead?
That would be a good idea, but I don't know why `setVehicleComponentScale` doesn't work with wheels. I guess it may have something to do with the `CVehicle` attribute that this PR exposes, or other GTA stuff.

## Added functions
```lua
float getVehicleWheelScale ( vehicle theVehicle )
```
Returns a decimal number that represents the scale multiplier in the vertical and horizontal axes that is applied for all the wheels of the vehicle. By default, 1.
```lua
bool setVehicleWheelScale ( vehicle theVehicle, float wheelScale )
```
Sets the wheel scale multiplier for all the wheels of the vehicle. A value of 1 draws the wheels with the default scale. A value of 0 is allowed, and it makes wheels disappear.

## Screenshots
![mta-screen_2020-08-28_23-25-35](https://user-images.githubusercontent.com/7822554/91616677-766d8580-e986-11ea-9a09-6c1bf343d010.png)
![mta-screen_2020-08-28_23-26-29](https://user-images.githubusercontent.com/7822554/91616680-78374900-e986-11ea-91e7-3bf8ae29c031.png)
![mta-screen_2020-08-28_23-30-10](https://user-images.githubusercontent.com/7822554/91616681-7a010c80-e986-11ea-9011-75df0dbec20e.png)

AlexTMjugador committed:
Add get/setVehicleWheelScale

These functions, related to issue #719, expose the wheel scale size
attribute of the CVehicle class to clientside Lua scripts.

qaisjp commented on Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp:
```diff
@@ -3664,6 +3664,18 @@ bool CStaticFunctionDefinitions::GetVehicleModelExhaustFumesPosition(unsigned sh
+{
+    Vehicle.SetWheelScale(fWheelScale);
+    return true;
+}
```
It looks like these can be removed :) (as well as in the header)

AlexTMjugador replied:
Sure 👍 

I'll remove them tomorrow, although I may supersede this PR with a better one soon™️.

qaisjp commented:
Mad props to you, @AlexTMjugador — the PR is really high quality and very descriptive! Thank you

AlexTMjugador committed:
Remove the new functions from CStaticFunctionDefinitions

AlexTMjugador committed:
Use override keyword in CVehicleSA new methods

This keyword guarantees that these methods override the virtual methods in the CVehicle class, as intended.

Einheit-101 commented:
I dont think this is a very good idea, maybe someone can find out why setVehicleComponentScale doesnt work with wheels and otherwise the server owner can still remodel his vehicle wheels, this is not a good fix for the missing vehicle wheel collision issue.

AlexTMjugador commented:
> I dont think this is a very good idea, maybe someone can find out why setVehicleComponentScale doesnt work with wheels and otherwise the server owner can still remodel his vehicle wheels, this is not a good fix for the missing vehicle wheel collision issue.

I agree with your point, @Einheit-101. This PR, alone, feels like a kinda bad fix for that issue. However, in combination with #1644, it may be better. What do you think?

ghost commented:
setVehicleComponentScale was designed to only modify the wheel geometry. The reason why it doesn't affect the collision is that there's no code to modify the collision. We shouldn't touch setVehicleComponentScale because it takes 3 parameters for scale, and that's absurd. We're dealing with wheels here, it should only take 1 parameter, but it's too late to fix the function.

Commit:
Merge branch 'master' into feature/wheel-scale

Dutchman101 commented:
@AlexTMjugador Please add the below wiki pages:

[https://wiki.multitheftauto.com/wiki/getVehicleWheelScale](https://wiki.multitheftauto.com/wiki/getVehicleWheelScale)
[https://wiki.multitheftauto.com/wiki/setVehicleWheelScale](https://wiki.multitheftauto.com/wiki/setVehicleWheelScale)

AlexTMjugador commented:
> @AlexTMjugador Please add the below wiki pages:
> 
> https://wiki.multitheftauto.com/wiki/getVehicleWheelScale
> https://wiki.multitheftauto.com/wiki/setVehicleWheelScale

Added, thanks for the reminder!
